tests/components/zoneminder/test_camera.py

```diff
@@ -0,0 +1,89 @@
+"""Binary sensor tests."""
+from zoneminder.monitor import Monitor
+from zoneminder.zm import ZoneMinder
+
+from homeassistant.components.zoneminder import const
+from homeassistant.config import async_process_ha_core_config
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PATH,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import DOMAIN as HASS_DOMAIN, HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.async_mock import MagicMock, patch
+from tests.common import MockConfigEntry
+
+
+async def test_async_setup_entry(hass: HomeAssistant) -> None:
+    """Test setup of camera entities."""
+    with patch(
+        "homeassistant.components.zoneminder.common.ZoneMinder", autospec=ZoneMinder
+    ) as zoneminder_mock:
+        monitor1 = MagicMock(spec=Monitor)
+        monitor1.name = "monitor1"
+        monitor1.mjpeg_image_url = "mjpeg_image_url1"
+        monitor1.still_image_url = "still_image_url1"
+        monitor1.is_recording = True
+        monitor1.is_available = True
+
+        monitor2 = MagicMock(spec=Monitor)
+        monitor2.name = "monitor2"
+        monitor2.mjpeg_image_url = "mjpeg_image_url2"
+        monitor2.still_image_url = "still_image_url2"
+        monitor2.is_recording = False
+        monitor2.is_available = False
+
+        zm_client: ZoneMinder = MagicMock(spec=ZoneMinder)
```

Comment:
Is it worth making the zm_client a shared fixture used with all of the tests?

Reply:
I don't think so. Each test sets up the client in unique ways. Abstracting it out to a fixture would only save ~2 lines per test. I don't think this savings is worth he extra test complexity.